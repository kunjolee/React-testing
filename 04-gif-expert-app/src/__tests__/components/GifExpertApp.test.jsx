import { fireEvent, render, screen } from '@testing-library/react';
import { GifExpertApp } from '../../GifExpertApp';

jest.mock('../../../constants', () => ({
    VITE_API_KEY: '1234567890',
}));

describe('Tests on <GifExpertApp />', () => {
    it('it should match the snapshot', () => {
        const { container } = render(<GifExpertApp />);
        expect(container).toMatchSnapshot();
    });

    it('it should add a new category', () => {
        render(<GifExpertApp />);

        const input = screen.getByRole('textbox');
        const form = screen.getByRole('form');
        fireEvent.change(input, { target: { value: 'NewCategory' } });
        fireEvent.submit(form);

        expect(screen.getByText('NewCategory')).toBeTruthy();
        expect(
            screen.getAllByRole('heading', { level: 3 }).length
        ).toBeGreaterThan(1);
    });
    it('it should not add a new category if it already exists', () => {
        render(<GifExpertApp />);

        const input = screen.getByRole('textbox');
        const form = screen.getByRole('form');

        fireEvent.change(input, { target: { value: 'One Punch' } });
        fireEvent.submit(form);
        expect(screen.getAllByRole('heading', { level: 3 }).length).toBe(1);
        expect(screen.getByText('One Punch')).toBeTruthy();
    });
});
