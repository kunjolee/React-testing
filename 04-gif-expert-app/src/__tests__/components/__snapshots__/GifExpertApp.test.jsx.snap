// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <GifExpertApp /> it should match the snapshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="aria-form"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
  <h3>
    One Punch
  </h3>
  <h2>
    Cargando...
  </h2>
</div>
`;
